zebra-node-services/src/rpc_client.rs

```diff
@@ -43,6 +43,26 @@ impl RpcRequestClient {
             .await
     }
 
+    /// Builds rpc request with a variable `content-type`. Used only for testing.
+    pub async fn call_with_content_type(
```

Comment:
Test-only methods should only be compiled in tests. This also requires an update to zebrad/Cargo.toml, I'll push that to this branch now.

```suggestion
    #[cfg(any(test, feature = "proptest-impl"))]
    pub async fn call_with_content_type(
```

Reply:
I don't think this change is needed so i reverted https://github.com/ZcashFoundation/zebra/pull/6885/commits/752be9486aab236ba0d3b2401d11a7ffc676f034 in https://github.com/ZcashFoundation/zebra/pull/6885/commits/d23175c56d9d572da3e5399343085724fc1513b5

The new methods are already under the rpc-client feature https://github.com/ZcashFoundation/zebra/blob/752be9486aab236ba0d3b2401d11a7ffc676f034/zebra-node-services/src/lib.rs#L6 and this is only called from tests: https://github.com/ZcashFoundation/zebra/blob/752be9486aab236ba0d3b2401d11a7ffc676f034/zebrad/Cargo.toml#L227